From oliver@neukum.org  Mon Mar 15 16:14:21 2010
From: Oliver Neukum <oliver@neukum.org>
Date: Sat, 27 Feb 2010 20:55:26 +0100
Subject: usb: cdc-wdm: Fix race between disconnect and debug messages
To: Greg KH <greg@kroah.com>
Cc: Rickard Bellini <rickard.bellini@ericsson.com>
Message-ID: <201002272055.26689.oliver@neukum.org>

From: Oliver Neukum <oliver@neukum.org>

dev_dbg() and dev_err() cannot be used to report failures
that may have been caused by a device's removal

Signed-off-by: Oliver Neukum <neukum@b1-systems.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/class/cdc-wdm.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/drivers/usb/class/cdc-wdm.c
+++ b/drivers/usb/class/cdc-wdm.c
@@ -435,11 +435,8 @@ retry:
 		spin_lock_irq(&desc->iuspin);
 
 		if (desc->rerr) { /* read completed, error happened */
-			int t = desc->rerr;
 			desc->rerr = 0;
 			spin_unlock_irq(&desc->iuspin);
-			dev_err(&desc->intf->dev,
-				"reading had resulted in %d\n", t);
 			rv = -EIO;
 			goto err;
 		}
@@ -477,8 +474,6 @@ retry:
 
 err:
 	mutex_unlock(&desc->lock);
-	if (rv < 0 && rv != -EAGAIN)
-		dev_err(&desc->intf->dev, "wdm_read: exit error\n");
 	return rv;
 }
 
